aiida/parsers/plugins/arithmetic/add.py

```diff
@@ -7,50 +7,37 @@
 # For further information on the license, see the LICENSE.txt file        #
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
+# pylint: disable=inconsistent-return-statements
+"""Parser for an `ArithmeticAddCalculation` job.
+
+.. warning::
+
+    This implementation is used directly in the documentation as a literal-include, which means that if any part of this
+    code is changed, the snippets in the file `docs/source/howto/codes.rst` have to be checked for consistency.
+"""
 from aiida.orm import Int
 from aiida.parsers.parser import Parser
 
 
 class ArithmeticAddParser(Parser):
-    """Parser for an ArithmeticAdd calculation."""
+    """Parser for an `ArithmeticAddCalculation` job."""
 
-    def parse(self, **kwargs):  # pylint: disable=inconsistent-return-statements
-        """Parse the contents of the output files retrieved in the `FolderData`."""
+    def parse(self, **kwargs):
+        """Parse the contents of the output files stored in the `retrieved` output node."""
         try:
             output_folder = self.retrieved
         except AttributeError:
             return self.exit_codes.ERROR_NO_RETRIEVED_FOLDER
 
         try:
-            with output_folder.open(self.node.get_attribute('output_filename'), 'r') as handle:
-                result = self.parse_stdout(handle)
-        except (OSError, IOError):
+            with output_folder.open(self.node.get_option('output_filename'), 'r') as handle:
+                result = int(handle.read())
+        except OSError:
             return self.exit_codes.ERROR_READING_OUTPUT_FILE
-
-        if result is None:
+        except ValueError:
             return self.exit_codes.ERROR_INVALID_OUTPUT
 
-        try:
-            allow_negative = self.node.inputs.settings.get_attribute('allow_negative', True)
```

Comment:
Was this attribute deprecated?

Reply:
It was not, but since this class is here purely for demonstration and internal testing purposes, I think we can remove it without going through a proper deprecation path.

Reply:
Ok, no problem then.